python2verilog/optimizer/optimizer.py

```diff
@@ -25,18 +25,21 @@ def is_dependent(expr: ir.Expression, var: str):
     raise TypeError(f"unexpected {type(expr)}")
 
 
-def backwards_replace(expr: ir.Expression, mapping: dict[ir.Expression, ir.Expression]):
+def backwards_replace(expr: ir.Expression, mapping: dict[ir.Var, ir.Expression]):
     """
     If the expression matches a key in the mapping, it is replaced with
     the corresponding value in the mapping.
 
+    Note: ignores exclusive vars in replacement process
+
     :return: a copy of the updated expression.
     """
     expr = copy.deepcopy(expr)
     if isinstance(expr, ir.Var):
-        for key in mapping:
-            if key.to_string() == expr.to_string():
-                return mapping[key]
+        if not isinstance(expr, ir.ExclusiveVar):
```

Comment:
Is there a reason why this if statement can't be combined with the above line?

Reply:
Yep

Reply:
Currently, if a variable is a `ir.Var` and not a `ir.ExclusiveVar`, no exception is thrown.